src/main/java/org/spongepowered/api/effect/Viewer.java

```diff
@@ -161,4 +161,60 @@ default void resetBlockChange(final Vector3i position) {
      */
     void resetBlockChange(int x, int y, int z);
 
+    /**
+     * Sends a client-only block breaking progress.
+     *
+     * <p>In vanilla breaking progress will not be rendered if player
+     * is further than 32 blocks from given position.<br>
+     * Sent breaking progress expires on client 400 ticks after receiving.</p>
+     *
+     * @param position The position
+     * @param progress The breaking progress from 0 to 1 (1 excluded)
+     */
+    default void sendBlockProgress(final Vector3i position, final float progress) {
+        Objects.requireNonNull(position, "position");
+        this.sendBlockProgress(position.x(), position.y(), position.z(), progress);
+    }
+
+    /**
+     * Sends a client-only block breaking progress.
+     *
+     * <p>In vanilla breaking progress will not be rendered if player
+     * is further than 32 blocks from given position.<br>
+     * Sent breaking progress expires on client 400 ticks after receiving.</p>
+     *
+     * @param x The x position
+     * @param y The y position
+     * @param z The z position
+     * @param progress The breaking progress from 0 to 1 (1 excluded)
+     */
+    void sendBlockProgress(int x, int y, int z, float progress);
+
+    /**
+     * Resets the client's view of the provided position to actual
+     * breaking progress.
+     *
+     * <p>This is useful for resetting what the client sees
+     * after sending a {@link #sendBlockProgress block progress}.</p>
+     *
+     * @param position The position
+     */
+    default void resetBlockProgress(final Vector3i position) {
+        Objects.requireNonNull(position, "position");
+        this.resetBlockProgress(position.x(), position.y(), position.z());
+    }
+
+    /**
+     * Resets the client's view of the provided position to actual
+     * breaking progress.
+     *
+     * <p>This is useful for resetting what the client sees
+     * after sending a {@link #sendBlockProgress block progress}.</p>
+     *
+     * @param x The x position
+     * @param y The y position
+     * @param z The z position
+     */
+    void resetBlockProgress(int x, int y, int z);
```

Comment:
Wdym it's impl detail. It's must have method if we make limits (from 0 to 1) for progress arg in sendBlockProgress

Reply:
Server doesn't have any knowledge of the current progress. It resets only progress that was send by `sendBlockProgress()` so the only knowledge server has is that some plugins used `sendBlockProgress()`

Reply:
I think `resetBlockProgress` is necessary because we need a way to remove the crack.
Notes:
1. setting progress to 0 does not remove, it just means you began cracking the block.
2. as discussed on discord, we will not track progress, however we need to keep somewhere a map of blockpos to destruction id